### ISSUE TITLE
Fix Async Search + Async DB Bug/Make Sure We Are Interacting With the DB Properly

### ISSUE THREAD

**Issue by notowen333** [privacy-tech-lab/privacy-pioneer#229]
I'm a bit bothered that often, when we create location evidence, that there are other elements that we should have flagged contained in the snippet.

For example maybe we have the `Tight Location` label but when you expand the snippet, you see the zip code from your watchlist, but there is no `Zip Code` label. I have a hunch that this is likely a cause of the async nature of the search functions and the DB, since both labels would be coming from the same request.

So, in short, **we need to sort out the async events in our analysis**.  Aside from this particular bug, it would be good to better understand how the async DB works, and make sure we are interacting with it properly.

Specifically, I think our practice of `.get` -> `await` -> `.set` creates the potential for problems.
